jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+'''
+Writes, reads and parses NMEA like messages from the onboard
+science nucleo to operate the science boxes and get relevant data
+'''
+import serial
+import asyncio
+# import Adafruit_BBIO.UART as UART  # for beaglebone use only
+import numpy as np
+import time
+from rover_common.aiohelper import run_coroutines
+from rover_common import aiolcm
+from rover_msgs import ThermistorData, MosfetCmd, SpectralData, \
+    NavStatus, ServoCmd, CarouselData, CarouselClosedLoopCmd, CarouselOpenLoopCmd, \
+    Heater, HeaterAutoShutdown
+
+
+class ScienceBridge():
+    def __init__(self):
+        # UART.setup("UART4")  # Specific to beaglebone
+        # maps NMEA msgs to their handler
+        self.NMEA_HANDLE_MAPPER = {
+            "SPECTRAL": self.spectral_handler,
+            "THERMISTOR": self.thermistor_handler,
+            "TRIAD": self.triad_handler,
+            "TXT": self.txt_handler,
+            "CAROUSEL": self.carousel_handler,
+            "HEATER": self.heater_state_handler,
+            "AUTOSHUTOFF": self.heater_shutoff_handler
+        }
+        self.last_openloop_cmd = -100
+        self.max_error_count = 20
+        self.sleep = .01
+
+    def __enter__(self):
+        '''
+        Opens a serial connection to the nucleo
+        '''
+        self.ser = serial.Serial(
+            # port='/dev/ttyS4',
+            port='/dev/ttyTHS1',
```

Comment:
can the port also be a config thing? or even just a config variable you set bc it changes from jetson to jetson and the beaglebone